File: Java/Human.java
/*Saya Muhammad Daffa Yusuf Fadhilah dengan NIM 2100543 mengerjakan evaluasi Latihan 
dalam mata kuliah Design Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti
yang telah dispesifikasikan. Aamiin. */

//Berikut class untuk 'Human' yang menjadi base class
public class Human {

    //Atribut dari 'Human'
    private String NIK;
    private String nama;
    private char jenis_kelamin;

    //Empty Construtor untuk 'Human'
    public Human() {
        NIK = "";
        nama = "";
        jenis_kelamin = '-';
    }
    
    //Parameter Construtor untuk 'Human'
    public Human(String NIK, String nama, char jenis_kelamin) {
        this.NIK = NIK;
        this.nama = nama;
        this.jenis_kelamin = jenis_kelamin;
    }


    //Setter dan Getter pada class 'Human'
    public void set_NIK(String NIK){
        this.NIK = NIK;
    }

    public void set_nama(String nama){
        this.nama = nama;
    }

    public void set_jenis_kelamin(char jenis_kelamin){
        this.jenis_kelamin = jenis_kelamin;
    }

    public String get_NIK(){
        return this.NIK;
    }

    public String get_nama(){
        return this.nama;
    }

    public char get_jenis_kelamin(){
        return this.jenis_kelamin;
    }
}

File: C++/Human.cpp
/*Saya Muhammad Daffa Yusuf Fadhilah dengan NIM 2100543 mengerjakan evaluasi Latihan 2
dalam mata kuliah Design Pemrograman Berorientasi Objek
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti
yang telah dispesifikasikan. Aamiin. */

//Include untuk penggunaan string
#include <iostream>
#include <string>

//Deklarasi std agar tidak perlu menulisnya kembali pada code
using namespace std;

//Membuat class Human
class Human
{
protected:

    //Menambahkan attribut yang dibutuhkan
    string NIK;
    string nama;
    char jenis_kelmain;

public:

    //Fungsi construct kosong
    Human(){
        this->NIK = "";
        this->nama = "";
        this->jenis_kelmain = '-';
    }

    //Fungsi construct dengan parameter
    Human(string NIK, string nama, char jenis_kelamin){
        this->NIK = NIK;
        this->nama = nama;
        this->jenis_kelmain - jenis_kelamin;
    }


    //setter dan getter tiap atribut yang ada pada kelas ini
    void set_NIK(string NIK){
        this->NIK = NIK;
    }

    void set_nama(string nama){
        this->nama = nama;
    }

    void set_jenis_kelamin(char jenis_kelamin){
        this->jenis_kelmain = jenis_kelamin;
    }

    string get_NIK(){
        return this->NIK;
    }

    string get_nama(){
        return this->nama;
    }

    char get_jenis_kelamin(){
        return this->jenis_kelmain;
    }

    ~Human(){

    }
};
